src/draw/lv_image_decoder.h

```diff
@@ -111,10 +112,20 @@ typedef struct _lv_image_decoder_t {
     lv_image_decoder_open_f_t open_cb;
     lv_image_decoder_get_area_cb_t get_area_cb;
     lv_image_decoder_close_f_t close_cb;
-    uint32_t cache_data_type;
+
+    lv_cache_free_cb_t cache_free_cb;
```

Comment:
From the name `lv_cache_free_cb_t`, it should belong to `lv_image_cache.c` where image cache instance will use this callback. I understand bin decoder is not unified with other decoders, so the free callback is not unified yet. Let's optimize it so we can use same `free_cb` for all decoders.